wandb/sdk/artifacts/storage_handlers/fsspec_file_handler.py

```diff
@@ -0,0 +1,146 @@
+"""Storage handler utilizing fsspec."""
+import os
+import time
+from typing import TYPE_CHECKING, Any, Optional, Sequence, Union
+from urllib.parse import ParseResult
+
+from wandb.errors.term import termlog
+from wandb.sdk.artifacts.artifact_file_cache import get_artifact_file_cache
+from wandb.sdk.artifacts.artifact_manifest_entry import ArtifactManifestEntry
+from wandb.sdk.artifacts.storage_handler import DEFAULT_MAX_OBJECTS, StorageHandler
+from wandb.sdk.lib import filesystem
+from wandb.sdk.lib.hashutil import B64MD5, md5_string
+from wandb.sdk.lib.paths import FilePathStr, StrPath, URIStr
+
+if TYPE_CHECKING:
+    import fsspec  # type: ignore
+
+    from wandb.sdk.artifacts.artifact import Artifact
+
+
+class FsspecFileHandler(StorageHandler):
+    """Handles a variety of different storage solutions."""
+
+    def __init__(self, scheme: Optional[str] = None) -> None:
+        """Track files or directories on a variety of filesystem.
+
+        For now, this handler supports oss (Alibaba Object Storage System)
+        A list of all available options can be found under the following link:
+        https://github.com/fsspec/filesystem_spec/blob/master/fsspec/registry.py
+        """
+        self._scheme = scheme or "oss"
```

Comment:
Here I would have liked to use `fsspec.registry.available_protocols()` however, this would require the import of fsspec being outside of the `if TYPE_CHECKING` block I assume... which causes some of the CI tests to fail

Reply:
You need to import `fsspec` anyway. See the comment below.